### ISSUE TITLE
make fails with -Wpedantic

### ISSUE THREAD

**Issue by melissawm** [coin-or/Dip#3]
Hi, 

I'm building Dip on Manjaro Linux with 6.3.1 and GNU Make 4.2.1 and after a successful configure step, I get this error when I run make:

../../../../Dip/src/dippy/DippyDecompAlgo.cpp:32:22: error: ISO C++ forbids converting a string constant to 'char*' [-Wpedantic] 
        pSolutionList);

(there are several more lines like this). I managed to build Dip by commenting the

coin_warn_cflags="-pedantic-errors $coin_warn_cflags"

lines from the configure script, however I don't know if this is supposed to happen.

**Comment by melissawm** [comment]
So I have no experience with C++ but after some research it appears this is an issue with >= Python 3.4. From the docs:

====

PyObject* PyObject_CallMethod(PyObject *o, const char *method, const char *format, ...)
Return value: New reference.
Call the method named method of object o with a variable number of C arguments. The C arguments are described by a Py_BuildValue() format string that should produce a tuple. The format may be NULL, indicating that no arguments are provided. Returns the result of the call on success, or NULL on failure. This is the equivalent of the Python expression o.method(args). Note that if you only pass PyObject * args, PyObject_CallMethodObjArgs() is a faster alternative.

** Changed in version 3.4: The types of method and format were changed from char *. **

====

Hope this helps!

Thanks.

**Comment by tkralphs** [comment]
Hi @melissawm, thanks for the report. We have turned off -pedantic-errors by default in the latest version of the COIN-OR build system because of some related problems with gcc 6, which seems to be getting more aggressive with adherence to the standard. I guess this problem will go away with the next release (although only in the sense that this warning will be ignored, although it would obviously be better to fix it instead). 

At the moment, DipPy does not officially support Python 3 and I have never personally tried to build it under anything but Python 2. Is it working under Python 3 then? If so, I'm pleasantly surprised :).

**Comment by melissawm** [comment]
Yes, it does appear to work with Python 3.6.0! make test is successful, at least.

Thanks for replying. 

Cheers!

**Comment by lucasemmoreira** [comment]
Hello!
I also have the same issue trying to build Dip. I've tried to comment the lines as suggested but the same errors arised. Another info (that I'm not sure if it is useful) is the final make lines given:

**Makefile:506: recipe for target '_dippy_la-DippyDecompAlgo.lo' failed
make[4]: *** [_dippy_la-DippyDecompAlgo.lo] Error 1
make[4]: Leaving directory '/home/bigode/AOC/dippy/Dip-0.92/Dip/src/dippy'
Makefile:698: recipe for target 'all-recursive' failed
make[3]: *** [all-recursive] Error 1
make[3]: Leaving directory '/home/bigode/AOC/dippy/Dip-0.92/Dip/src'
Makefile:489: recipe for target 'all' failed
make[2]: *** [all] Error 2
make[2]: Leaving directory '/home/bigode/AOC/dippy/Dip-0.92/Dip/src'
Makefile:710: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/home/bigode/AOC/dippy/Dip-0.92/Dip'
Makefile:334: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1**

I am not sure what kind of information is enough to give but I'll try to give the basics:
I am using Ubuntu 17.04.
I've downloaded Dip from the repo using git.
My make version is 4.1
The configure was successful.

I've followed the instructions in https://projects.coin-or.org/Dip at Download and Install.
I am not sure how I can progress from here.
I appreciate any help!

**Comment by tkralphs** [comment]
If you could provide more of the output showing the actual compiler error and also say what compiler and version you are using, that would help. I'll see if I can replicate the error.

**Comment by lucasemmoreira** [comment]
Thank you for the fast response!
I did the process again, commenting the two lines as suggested.
configure output: https://gist.github.com/lucasemmoreira/df73b710c9a2cceabae939ba5eb2891b
make output: https://gist.github.com/lucasemmoreira/4de7a61e245905e8e16c40b8c1f320aa
Not sure if it is still necessary to give information of the compiler. If it is, let me know and I'll let you kow.
Thank you again.

**Comment by lucasemmoreira** [comment]
I have managed to go through thanks to your response. But instead of 
```
char * arg1 = "chooseBranchSet", arg2 = "O";
PyObject* pResult = PyObject_CallMethod(m_pProb, arg1, arg2,  pSolutionList);
```
I used
```
   char arg1[] = "chooseBranchSet";
   char arg2[] = "O";
   PyObject* pResult = PyObject_CallMethod(m_pProb, arg1, arg2,  pSolutionList);

```
and it worked to build Dip!
I also had to make similar changes in `DippyDecompApp.cpp` and `DippyPythonUtils.cpp`. All the changes I made was on demmand, i.e, only on the lines `make` raised an error. Not sure if this is enough for a pull request, but if want I can make one.

Again, thank you for your help!

**Comment by tkralphs** [comment]
Yes, it would be great if you could make a pull request. I can always improve it as needed, but good to have a start. Thanks!
